scalafx-demos/src/main/scala/scalafx/imaginej/JumpingFrogsPuzzle.scala

```diff
@@ -209,7 +209,7 @@ class Model(var optionalFrogMap: Map[Int, Option[Frog]]) {
 
   private def positionSingleton(frog: Frog) =
     for {
-      (i, Some(`frog`)) <- optionalFrogMap
+      case (i, Some(`frog`)) <- optionalFrogMap
```

Comment:
This change does not improve the code. Should be reverted.

Reply:
When I revert the code and run it with `scala 3.4.2`, I get this:
```scala
-- Error: C:\Users\philwalk\workspace\scalafx-work\scalafx-demos\src\main\scala\scalafx\imaginej\JumpingFrogsPuzzle.scala:212:10 ---------------------------------------------
212 |      (i, Some(`frog`)) <- optionalFrogMap
    |          ^^^^^^^^^^^^
    |          pattern's type Some[scalafx.imaginej.Frog] is more specialized than the right hand side expression's type Option[scalafx.imaginej.Frog]
    |
    |          If the narrowing is intentional, this can be communicated by adding the `case` keyword before the full pattern,
    |          which will result in a filtering for expression (using `withFilter`).
    |          This patch can be rewritten automatically under -rewrite -source 3.2-migration.
1 error found
Errors encountered during compilation
```

Reply:
Thanks for the explanation. ScalaFX will not yet build with option `-source 3.2-migration`, but let make step in that direction.